mindmeld/components/dialogue.py

```diff
@@ -957,7 +977,25 @@ def __call__(self, request, responder):
                 self._prompt_turn = True
 
         # Finish slot-filling and return to handler
-        return self._end_slot_fill(request, responder)
+        return self._end_slot_fill(request, responder, self._app.async_mode)
+
+    def call_sync(self, request, responder):
+        """The slot-filling call for synchronous apps
+
+        Args:
+            request (Request): The request object.
+            responder (DialogueResponder): The responder object.
+        """
+        self.__call__(request, responder)
+
+    async def call_async(self, request, responder):
+        """The slot-filling call for asynchronous apps
+
+        Args:
+            request (Request): The request object.
+            responder (DialogueResponder): The responder object.
+        """
+        self.__call__(request, responder)
```

Comment:
@tmehlinger: my knowledge of async is a bit spotty here but do we need an await statement here as well?
```suggestion
        await self.__call__(request, responder)
```